concrete/elements/block_header_edit.php

```diff
@@ -1,20 +1,36 @@
 <?php
 defined('C5_EXECUTE') or die('Access Denied.');
 
-/* @var Concrete\Core\Block\Block $b */
```

Comment:
Why did you remove this line?

Reply:
It's there https://github.com/concrete5/concrete5/pull/6279/files#diff-ae6b89a5607f08f640764fd8541197dfR4

Reply:
The specs says that we should have the type before the variable name (that is: `/* @var Concrete\Core\Block\Block $b */` is correct, `/* @var $b Concrete\Core\Block\Block */` is wrong).
See http://docs.phpdoc.org/references/phpdoc/tags/var.html

Reply:
I don't have arguments for that, but for Netbeans the format you mentioned works with Class property
```php
class Foo
{
  /** @var string|null Should contain a description */
  protected $description = null;
}
```
BUT doesn't work inside code and we should use this one
```php
/* @var $b Concrete\Core\Block\Block */
```

Reply:
See
https://blogs.oracle.com/netbeansphp/defining-a-variable-type-in-comment-ii

Reply:
If NetBeans doesn't follow the specs, we should fix NetBeans instead of the code... I'd file a bug report at https://netbeans.org/bugzilla

Reply:
ok i'll fix the code

Reply:
You just need to write the comments as following:
`/* @var $myObject \Application\What\You\Want */`
Then it should work on NetBeans to get the object properties.

Reply:
Thanks i've moved to PhpStorm ;-)

Reply:
> /* @var $myObject \Application\What\You\Want */

@1stthomas As [I wrote above](https://github.com/concrete5/concrete5/pull/6279#discussion_r160343118), the specs say that the correct way of describing variables is to place the type *before* the variable name. If NetBeans behaves differently, we should fix NetBeans, not the code... If you use NetBeans you should submit a bug report at https://netbeans.org/bugzilla

Reply:
@mlocati: This one: https://blogs.oracle.com/netbeansphp/defining-a-variable-type-in-comment-ii is not a spec, it is a blog post from 2009. I could not find a netbeans spec for this topic.
Typing the way as found on my previous post works for NetBeans >= 8.0.2 . Therfor it is not an issue of the spec, it is an issue of `vdoc`, because this generates still the buggy comment. But this syntax does not work.

I did not wanted to change the c5 to the buggy netbeans syntax. I just wanted to let you know, what the real problem is. But I agree with you, that it should be reported on https://netbeans.org/bugzilla ;)

Reply:
> @mlocati: This one: https://blogs.oracle.com/netbeansphp/defining-a-variable-type-in-comment-ii is not a spec, it is a blog post from 2009. I could not find a netbeans spec for this topic.

@1stthomas Indeed: [this one](http://docs.phpdoc.org/references/phpdoc/tags/var.html) is the spec I wrote [here](https://github.com/concrete5/concrete5/pull/6279#discussion_r160343118).

> Typing the way as found on my previous post works for NetBeans >= 8.0.2 . Therfor it is not an issue of the spec, it is an issue of vdoc, because this generates still the buggy comment. But this syntax does not work.

I didn't say it was an issue of the spec: I wrote it's an issue of NetBeans. And I absolutely don't know what `vdoc` is :wink: